Assets/Mirror/Editor/Weaver/Writers.cs

```diff
@@ -226,22 +226,20 @@ static MethodDefinition GenerateArrayWriteFunc(TypeReference variable, int recur
             worker.Append(worker.Create(OpCodes.Brtrue, labelNull));
 
             worker.Append(worker.Create(OpCodes.Ldarg_0));
-            worker.Append(worker.Create(OpCodes.Ldc_I4_0));
-            worker.Append(worker.Create(OpCodes.Call, Weaver.NetworkWriteUInt16));
+            worker.Append(worker.Create(OpCodes.Ldc_I4_M1));
+            worker.Append(worker.Create(OpCodes.Call, Weaver.NetworkWriterWritePackedInt32));
             worker.Append(worker.Create(OpCodes.Ret));
 
             // setup array length local variable
             worker.Append(labelNull);
             worker.Append(worker.Create(OpCodes.Ldarg_1));
             worker.Append(worker.Create(OpCodes.Ldlen));
-            worker.Append(worker.Create(OpCodes.Conv_I4));
-            worker.Append(worker.Create(OpCodes.Conv_U2));
```

Comment:
why is this not needed anymore?

Reply:
because the length variable is now int,  and not ushort,  so I don't have to convert them anymore.